runtests.py

```diff
@@ -451,6 +453,8 @@ def build_project(args):
             env['LDFLAGS'] = " ".join(cvars['LDSHARED'].split()[1:]) + ' --coverage'
 
     cmd += ["build"]
+    if args.compiler:
+        cmd += [f"--compiler={args.compiler}"]
```

Comment:
Further down there's a line:
```
compiler = env.get('CC') or cvars.get('CC', '')
```
which then gets used to add some flags. Could you make that use `args.compiler` if defined?

Reply:
Added below.